File: b2c-parent/b2c-core/src/main/java/org/javamaster/b2c/core/enums/ReadEnum.java
package org.javamaster.b2c.core.enums;

/**
 * @author yudong
 * @date 2019/6/10
 */
public enum ReadEnum implements EnumBase {
    UNREAD(1, "未读"),
    READ(2, "已读"),
    ;
    private final int code;
    private final String msg;


    ReadEnum(int code, String msg) {
        this.code = code;
        this.msg = msg;
    }


    @Override
    public int getCode() {
        return code;
    }

    @Override
    public String getMsg() {
        return msg;
    }

}

File: fragmentlearning/app/src/main/java/org/javamaster/fragmentlearning/enums/ReadEnum.kt
package org.javamaster.fragmentlearning.enums

/**
 * @author yudong
 * @date 2019/9/20
 */
enum class ReadEnum constructor(private val code: Int, private val msg: String) : EnumBase {
    UNREAD(1, "未读"),
    READ(2, "已读");


    override fun getCode(): Int {
        return code
    }

    override fun getMsg(): String {
        return msg
    }

}
